components/sup/src/topology/mod.rs

```diff
@@ -331,10 +331,12 @@ fn run_internal<'a>(sm: &mut StateMachine<State, Worker<'a>, SupError>,
                         // Write the configuration, and restart if needed
                         if try!(service_config.write(&package)) {
                             try!(package.copy_run(&service_config));
-                            try!(package.reconfigure(&service_config));
                             outputln!("Restarting because the service config was updated via the \
                                        census");
-                            restart_process = true;
+                            let has_reconfigure = try!(package.reconfigure(&service_config));
+                            if !has_reconfigure {
```

Comment:
I like how you've pulled out `has_reconfigure` instead of lumping it all together on one line.